src/assets/scripts/client/aircraft/FlightManagementSystem/LegModel.js

```diff
@@ -523,14 +523,15 @@ export default class LegModel {
      * @private
      */
     _findMinOrMaxAltitudeInProcedure(isMaximum = false) {
-        const altitudeRestrictionList = _map(this.waypointCollection, (waypoint) => waypoint.altitudeRestriction);
-
         if (isMaximum) {
-            return Math.max(...altitudeRestrictionList);
+            const maximumAltitudes = _map(this.waypointCollection, (waypoint) => waypoint.altitudeMaximum);
```

Comment:
It is possible to do this whole thing in a single `return`, but I'm glad you split this up.  Nice work! 👍